Examples/MAX32665/LP/main.c

```diff
@@ -205,12 +205,65 @@ void prepForDeepSleep(void)
     /* Prevent SIMO soft start on wakeup */
     MXC_LP_FastWakeupDisable();
 
-    /* Enable VDDCSWEN=1 prior to enter backup/deepsleep mode */
+    /* Enable VDDCSWEN=1 prior to enter DEEPSLEEP */
     MXC_MCR->ctrl |= MXC_F_MCR_CTRL_VDDCSWEN;
 
+    /* SIMO softstart workaround: clock 30KHz/8 for DEEPSLEEP, 30KHz/1 in ACTIVE */
+    *(volatile int *)0x40005434 = 1;
+    *(volatile int *)0x40005440 = (*(volatile int *)0x40005440 & (~(0x3 << 24))) | (0x2 << 24);
+    *(volatile int *)0x40005444 = (*(volatile int *)0x40005444 & (~(0x3 << 6))) | (0x2 << 6);
+
     switchToHIRCD4();
 
-    MXC_SIMO_SetVregO_B(810); /* Reduce VCOREB to 0.81v */
+    /* Wait for VCOREB to be ready */
+    while (!(MXC_SIMO->buck_out_ready & MXC_F_SIMO_BUCK_OUT_READY_BUCKOUTRDYB)) {}
+
+    /* Lower VregB to reduce power consumption */
+    MXC_SIMO_SetVregO_B(900);
+
+    /* Move VCORE switch to VCOREB (< VCOREA) */
+    MXC_MCR->ctrl = (MXC_MCR->ctrl & ~(MXC_F_MCR_CTRL_VDDCSW)) | (0x2 << MXC_F_MCR_CTRL_VDDCSW_POS);
+
+    /* Wait for VCOREA ready.  Should be ready already */
+    while (!(MXC_SIMO->buck_out_ready & MXC_F_SIMO_BUCK_OUT_READY_BUCKOUTRDYC)) {}
+}
+
+void prepForBackup(void)
+{
+    MXC_ICC_Disable();
```

Comment:
Should the prepForBackup and prepForDeepSleep functions be combined into one with an input parameter to select the mode?

Reply:
Got it. Will require a bit more work on that.

Reply:
Moved SIMO workaround code into PeriphDrivers. Still have separate prep functions for deep sleep and backup mode. I feel it is simpler and easier to use this way; not clear to me if we gain anything by combining into one function. There will be some overhead for passing a parameter and extra if/else inside the function.